### DISCUSSION TITLE
Using Scss modules in Next 13

### DISCUSSION THREAD

**Discussion by Cascade-8** [vercel/next.js#42028] [Help]
Wanted Behaviour: If i change some code in my scss module, the old class should be removed and the new one should get applied.

Current Behaviour: Both, the old and the new class are still existing, so it is impossible to see changes immediatly.

Same config etc. working fine in next <13
In the example below, I started to add a padding,  then changed it to a margin. But on localhost:3000 it rendered both, the padding and the margin. I either have to reload the full page everytime i change something, or manually deselect the css attributes.

![449170a822655f13b44233650913b843](https://user-images.githubusercontent.com/81745896/198461345-a3e34d5b-d921-4dcb-9888-126c0d4da201.png)

Edit: Doesn't work on client side Components, styles do not even get applied.

Is this a bug from the fast reload?

**Comment by icyJoseph** [comment]
I am also struggling a bit to grasp predictability when CSS-in-JS and CSS Modules are used. 

I am seeing all from missing styles, to flashes of un-styled content. I think the styling bit is the thing where most of us are struggling at the moment. The nested divs also make some layouts break... etc

Most notably, using `use client` directives really throws off the styling, and it never applies, unless I import the module up in the layout level. Perhaps it is meant to be like that but it is, nonetheless, hard to grasp right now.

![Screenshot 2022-10-28 at 09 26 02](https://user-images.githubusercontent.com/21013447/198528859-9c94ef4b-1d34-4ac2-b3e2-50c5f7ab1f4e.png)

In this case that component uses the client directive to be able to use `usePathname` and mark active links, even when there's some dynamic pages being generated further down the tree. When I give up that feature and allow this component to be server component, then the style is good. Though I still see flashes of un-styled content when navigating to other pages.

**Reply by Cascade-8** [reply]
Do you have the same issue with old css classes not beeing deleted? It is literally impossible to test your stylings, since the old classes still apply their css.

Yes I found out aswell, that you have to import the module in the layout.tsx

**Reply by icyJoseph** [reply]
I think the HMR is doing something funky. I do see a bunch of scratched out CSS :/

**Comment by JarrodMFlesch** [comment]
I have this exact issue, it's brings my computer to a halt. This issue makes development very difficult.

**Comment by JarrodMFlesch** [comment]
I opened an issue [here](https://github.com/vercel/next.js/issues/42702) with a reproducible stackblitz example. No need for duplicate comments and +1's here
